homeassistant/components/http/auth.py

```diff
@@ -34,7 +35,7 @@ def async_sign_path(hass, refresh_token_id, path, expiration):
         secret,
         algorithm="HS256",
     )
-    return f"{path}?{SIGN_QUERY_PARAM}=" f"{encoded.decode()}"
+    return f"{quote(path)}?{SIGN_QUERY_PARAM}=" f"{encoded.decode()}"
```

Comment:
This is fixing it at the wrong place? Shouldn't it be quoted by the function that passes it in?

Also please add some tests

Reply:
OK, but then we need to unquote the quoted path before signing. Maybe that's cleaner?

Reply:
Thinking about it, shouldn't the path get quoted wherever it is outputted ? It just seems very weird to do it here, as it means that if the path is conditionally signed, in the other case it's not quoted at all -> meaning it would break.

Reply:
So this method should do nothing, and whoever is generating this path, needs to quote things ?

Reply:
Updated so quoting is done by the caller, and we instead unquote the path to be signed.